crates/swc_ecma_transforms_compat/src/es2015/regenerator/hoist.rs

```diff
@@ -54,6 +56,27 @@ impl Hoister {
             exprs,
         })
     }
+
+    fn function_to_expr(&mut self, mut func: FnDecl) -> Expr {
+        func.visit_mut_children_with(self);
+
+        let FnDecl {
+            ident, function, ..
+        } = func;
+
+        self.vars.push(ident.clone());
+
+        AssignExpr {
+            span: function.span,
+            left: PatOrExpr::Pat(Box::new(ident.clone().into())),
+            op: op!("="),
+            right: Box::new(Expr::Fn(FnExpr {
+                ident: private_ident!(ident.span, format!("_{}", ident.sym)).into(),
```

Comment:
I think this will break a function that calls itself if it reaches here.
Can you add a test for it? (Or some comments describing why the function declarations cannot reach here)

You can fix it by using same syntax context as original one.
`Ident::new(&format!("_{}", ident.sym), ident.span)` will fix the issue.